core/src/main/scala/org/ensime/core/Project.scala

```diff
@@ -180,7 +173,10 @@ class Project(
 
     // added here to prevent errors when client sends this repeatedly (e.g. as a keepalive
     case ConnectionInfoReq =>
-      sender() ! ConnectionInfo()
+      if (serverConfig.legacy.connectionInfoReq)
+        sender() ! ConnectionInfo()
+      else
+        sender() ! AsyncConnectionInfo()
```

Comment:
this should always return the `ConnectionInfo`.

The `Greeting` should be sent as soon as the actor becomes `handleRequests`, which could be on startup if the legacy setting is disabled.

Reply:
Ah, ok. But to whom do I send it?